File: project/src/main/java/com/example/project/dto/UserDto.java
package com.example.project.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;


public class UserDto {

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class UserInfo {
        private String id;
        private String firstName;
        private String lastName;
        private String email;
        private LocalDateTime registrationDate;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class UpdateRequest {
        @NotBlank(message = "Имя обязательно")
        @Size(min = 1, max = 50, message = "Имя должно быть от 1 до 50 символов")
        private String firstName;

        @NotBlank(message = "Фамилия обязательна")
        @Size(min = 1, max = 50, message = "Фамилия должна быть от 1 до 50 символов")
        private String lastName;
    }
}

File: front/src/dto/userDto.js
// DTO для ответа с информацией о пользователе, соответствует backend `UserDto.UserInfo`
export function UserInfoDto(data) {
    if (!data) return null;
    return {
        id: data.id,
        firstName: data.firstName,
        lastName: data.lastName,
        email: data.email,
        registrationDate: data.registrationDate,
    };
}

export function UpdateUserRequestDto(firstName, lastName) {
    return {
        firstName: firstName,
        lastName: lastName,
    };
}

export function PageDto(data, itemMapper = (item) => item) {
    if (!data) return {
        content: [],
        totalElements: 0,
        totalPages: 0,
        number: 0,
        size: 0,
    };
    return {
        content: Array.isArray(data.content) ? data.content.map(itemMapper) : [],
        totalElements: data.totalElements || 0,
        totalPages: data.totalPages || 0,
        number: data.number || 0,
        size: data.size || 0,
    };
}
